// ngx-drag-scroll/esm2022/lib/ngx-drag-scroll-item.mjs
import { Directive, ElementRef, Input, HostBinding, Inject } from '@angular/core';
import * as i0 from "@angular/core";
class DragScrollItemDirective {
    get dragDisabled() {
        return this._dragDisabled;
    }
    set dragDisabled(value) {
        this._dragDisabled = value;
    }
    constructor(elementRef) {
        this.display = 'inline-block';
        this._dragDisabled = false;
        this._elementRef = elementRef;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: DragScrollItemDirective, deps: [{ token: ElementRef }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.0.0", type: DragScrollItemDirective, isStandalone: true, selector: "[drag-scroll-item]", inputs: { dragDisabled: ["drag-disabled", "dragDisabled"] }, host: { properties: { "style.display": "this.display" } }, ngImport: i0 }); }
}
export { DragScrollItemDirective };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: DragScrollItemDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[drag-scroll-item]',
                    standalone: true
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef, decorators: [{
                    type: Inject,
                    args: [ElementRef]
                }] }]; }, propDecorators: { display: [{
                type: HostBinding,
                args: ['style.display']
            }], dragDisabled: [{
                type: Input,
                args: ['drag-disabled']
            }] } });
